### PR TITLE
Improve CMake build

### PR DESCRIPTION
**Pull request by isaachier**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -73,15 +68,99 @@ set(_SRCS src/regint.h src/regparse.h src/regenc.h src/st.h
...
+    EXPORT "${TARGETS_EXPORT_NAME}"
+    NAMESPACE "${namespace}"
+    DESTINATION "${config_install_dir}"
+)
+
+# Documentation
+install(FILES doc/API doc/API.ja doc/RE doc/RE.ja doc/FAQ doc/FAQ.ja
+              doc/CALLOUTS.BUILTIN doc/CALLOUTS.BUILTIN.ja
+              doc/UNICODE_PROPERTIES
+        DESTINATION ${CMAKE_INSTALL_DOCDIR})
```
Comment (ruslo) [inline CMakeLists.txt]: I expecting something like `${CMAKE_INSTALL_DOCDIR}/${PROJECT_NAME}` here.

File: CMakeLists.txt [hunk 2]
```diff
@@ -73,15 +68,99 @@ set(_SRCS src/regint.h src/regparse.h src/regenc.h src/st.h
...
+
+# Other files
+install(FILES AUTHORS COPYING HISTORY README.md
+        DESTINATION ${CMAKE_INSTALL_DATADIR})
```
Comment (ruslo) [inline CMakeLists.txt]: Same, may be `${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME}`?

**Review by ruslo — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 3]
```diff
@@ -151,15 +151,15 @@ install(
     DESTINATION "${config_install_dir}"
 )
 
-# Documentation
+# Documentation (uses onig not oniguruma for directory)
```
Comment (isaachier) [inline CMakeLists.txt]: See original `dist.cmake` for reasoning here.

**Review by ruslo — APPROVED** [review]

**Comment by ruslo** [comment]
Released: https://github.com/hunter-packages/oniguruma/releases/tag/v6.8.1-p0

**Pull request merged**